guest-js/index.ts

```diff
@@ -1,11 +1,126 @@
 import { invoke } from '@tauri-apps/api/core';
 
+// ============================================================================
+// Types
+// ============================================================================
+
+/**
+ * Single metadata item extracted from MP4 `ilst` box.
+ */
+export interface Meta {
+   /** Raw fourcc key, using `@` for 0xA9 (e.g., "@nam"). */
+   key: string;
+   /** Friendly mapped name (e.g., "Title", "Artist", or "Unknown"). */
+   name: string;
+   /** Extracted value (UTF-8, trimmed of null padding). */
+   value: string;
+}
+
 /**
- * Says hello from the Media Parser plugin
- * @param name - The name to greet
- * @returns A greeting message
+ * Metadata extracted from an MP4 file.
  */
-export async function hello(name: string): Promise<string> {
-   return await invoke<string>('plugin:media-parser|hello', { name });
+export interface Metadata {
```

Comment:
The Rust `Metadata` struct has a format field that's serialized, but the TypeScript interface doesn't include it.